google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -26,7 +26,12 @@ class Results
         ##
         # Indicates the field names and types for the rows in the returned data.
         #
-        # @return [Hash] The types of the returned data.
+        # @param [Boolean] pairs Allow the types to be represented as a nested
+        #   Array of pairs rather than a Hash. This is useful when results have
+        #   duplicate names. The default is `false`.
```

Comment:
I don't think raising here is the best approach. A Hash is the most common data structure for querying a database, and I think most Rubyists will expect duplicate names to be squashed. As long as we give them a way to work with duplicate names I think we should be fine.

I have another change coming that will improve how missing names are handled. The duplicate names will still be an issue.